src/group/managed_group/mod.rs

```diff
@@ -579,7 +575,12 @@ impl<'a> ManagedGroup<'a> {
         let existing_key_package = tree.own_key_package();
         let key_package_bundle = match key_package_bundle_option {
             Some(kpb) => kpb,
-            None => KeyPackageBundle::from_rekeyed_key_package(existing_key_package),
+            None => {
+                let mut key_package_bundle =
+                    KeyPackageBundle::from_rekeyed_key_package(existing_key_package);
```

Comment:
Shouldn't we handle the signature internally in the `from_rekeyed_key_package` function? It seems like this is something we want to do every time we use that function.

Reply:
I was going to go for that, but we also use the function when we do full Commits. There we need to first add the `ParentHashExtension` to the `KeyPackage` and then sign. So we can either split it into two functions or just remember to sign later.
PS: I think it used to be called `..._unsigned()` but I removed it as a response to PR reviews. We could also re-introduce that, but then the name would be awfully long.

Reply:
I see. Then let's leave it as it is for now.